File: 2.2.py
import time 
# 定义函数fibonacci，参数n
n = 100
# 定义变量t1,t2，分别赋值为0和1
t1, t2 = 0, 1
# 打印输出Fibonacci Series
print("Fibonacci Series: ", end=" ")
# 记录函数开始时间
start = time.perf_counter()
# 循环n次，每次打印输出t1，并将t1和t2的值进行交换
for i in range(n):
    print(t1, end=" ")
    nextTerm = t1 + t2
    t1 = t2
    t2 = nextTerm

# 记录函数结束时间
end = time.perf_counter()

# 计算函数运行时间
time_spent = end - start
# 打印输出函数运行时间
print("\nTime taken: ", time_spent, " seconds")

File: 2.2.c
#include <stdio.h>
#include <time.h>

int main() {
    // 定义变量n，表示要输出多少个斐波那契数列
    int n = 100, i, t1 = 0, t2 = 1, nextTerm;
    // 输出字符串
    printf("Fibonacci Series: ");

    // 记录开始时间
    clock_t start = clock();
    // 循环输出斐波那契数列
    for (i = 1; i <= n; ++i) {
        // 输出斐波那契数列的第i个数
        printf("%d, ", t1);
        // 计算下一个斐波那契数列的值
        nextTerm = t1 + t2;
        // 将t1的值赋给t2
        t1 = t2;
        // 将nextTerm的值赋给t2
        t2 = nextTerm;
    }
    // 记录结束时间
    clock_t end = clock();

    // 计算程序运行时间
    double time_spent = (double)(end - start) / CLOCKS_PER_SEC;
    // 输出程序运行时间
    printf("\nTime taken: %f seconds\n", time_spent);

    return 0;

    /*多行的代码
    要如何注释掉呢*/
}